gherkin/php/src/AstNode.php

```diff
@@ -62,8 +62,14 @@ public function getSingle(string $expectedType, RuleType $ruleType, ?object $def
         return $items[0] ?? $defaultValue;
     }
 
-    /** needed for non-object return */
-    public function getSingleUntyped(RuleType $ruleType, mixed $defaultValue = null): mixed
+    /**
+     * needed for non-object return
+     *
+     * @param null|object $defaultValue
```

Comment:
This isn't correct, this method is here because not everything in PHP is an object (it's the non-object version of getSingle) so `mixed` is the correct type here

Additionally the use of `S` as a template below isn't meaningful as `S` isn't defined anywhere

I think this whole docblock can be reverted as it's not causing the issue